pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -660,7 +661,22 @@ protected BrokerResponse handleRequest(long requestId, String query, SqlNodeAndO
       List<ProcessingException> exceptions = new ArrayList<>();
       if (numUnavailableSegments > 0) {
         String errorMessage;
-        if (numUnavailableSegments > MAX_UNAVAILABLE_SEGMENTS_TO_PRINT_IN_QUERY_EXCEPTION) {
+        if (((realtimeTableConfig != null && offlineTableConfig != null)
```

Comment:
It is possible that one table is disabled and the other table has some unavailable segments. To handle all scenarios, we can add 2 boolean fields `offlineTableDisabled` and `realtimeTableDisabled` in addition to `unavailableSegments`. We can skip adding unavailable segments when a table is disabled

Reply:
Other than consolidating the `isTableDisabled` calls into local variables, does the logic need to change at all? In this version, I think I am skipping adding unavailable segments if either the offline or realtime table is disabled. Feel free to let me know if that's wrong.

Reply:
Say real-time table is disable, and 3 segments are unavailable in offline table, we want to log them both. Currently it will be logged only as `offline table disabled` but not with the unavailable segments.

In order to achieve that, we should check whether the table is disabled in line 624 and 641, and track its unavailable segments only when it is enabled

Reply:
Got it, that makes sense. I had to move this check inside the if statements referenced at 624 and 641 because adding this extra condition interfered with other behavior. I've gone ahead and made those changes to the PR to have this intended behavior. Let me know if that looks good!